### PR TITLE
Move ActivatorUtilities to Microsoft.Framework.DependencyInjection.Interfaces

### PR DESCRIPTION
**Pull request by Praburaj**
...tion.Interfaces on Http.Extensions

Fixes: https://github.com/aspnet/HttpAbstractions/issues/228

@davidfowl

### CODE REVIEW

File: src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.Framework.DependencyInjection.Tests")]
```
Comment (pranavkm) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs]: Why is this needed?

File: src/Microsoft.Framework.DependencyInjection.Interfaces/project.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
                 "System.Diagnostics.Debug": "4.0.10-beta-*",
                 "System.Globalization": "4.0.10-beta-*",
                 "System.Linq": "4.0.0-beta-*",
+                "System.Linq.Expressions": "4.0.10-beta-*",
```
Comment (pranavkm) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/project.json]: Could you remove this dependency from `Microsoft.Framework.DependencyInjection\project.json` if it exists?

File: test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@
 using Microsoft.Framework.DependencyInjection.Tests.Fakes;
 using Xunit;
 
-namespace Microsoft.Framework.DependencyInjection.Tests
+namespace Microsoft.Framework.DependencyInjection.Interfaces.Tests
```
Comment (pranavkm) [inline test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs]: Unrelated to this pr, but drop the `.Tests` portion from the namespace.

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (Praburaj) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs]: The tests uses the `Resources` defined in the assembly. So this is needed.

File: src/Microsoft.Framework.DependencyInjection.Interfaces/project.json (hunk 2 above)
Comment (Praburaj) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/project.json]: Let me check. I'll remove it if possible.

File: test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs (hunk 3 above)
Comment (Praburaj) [inline test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs]: Hmm. Interesting. All other tests had this namespace.

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs]: really....

Comment (Praburaj) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/Properties/AssemblyInfo.cs]: yeah we have the same on `Microsoft.Framework.DependencyInjection` assembly as well.

**Comment by Praburaj** [comment]
merged : https://github.com/aspnet/DependencyInjection/commit/702e9a7af37a20e1276521250d68c739778606b2
